Partial backport of the following commits to fix ARM32 build
fails with clang (https://crbug.com/1042348):
https://github.com/openssl/openssl/commit/313e6ec11fb8a7bda1676ce5804bee8755664141
https://github.com/openssl/openssl/commit/11208dcfb9105e8afa37233185decefd45e89e17

diff --git a/crypto/bn/asm/armv4-mont.pl b/crypto/bn/asm/armv4-mont.pl
index 8961180..cc87289 100644
--- a/crypto/bn/asm/armv4-mont.pl
+++ b/crypto/bn/asm/armv4-mont.pl
@@ -83,12 +83,13 @@ $code=<<___;
 
 .align	5
 bn_mul_mont:
+.Lbn_mul_mont:
 	ldr	ip,[sp,#4]		@ load num
 	stmdb	sp!,{r0,r2}		@ sp points at argument block
 #if __ARM_MAX_ARCH__>=7
 	tst	ip,#7
 	bne	.Lialu
-	adr	r0,bn_mul_mont
+	adr	r0,.Lbn_mul_mont
 	ldr	r2,.LOPENSSL_armcap
 	ldr	r0,[r0,r2]
 	tst	r0,#1			@ NEON available?
